src/renderers/shared/fiber/ReactFiberCollection.js

```diff
@@ -0,0 +1,123 @@
+/**
+ * Copyright 2013-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @providesModule ReactFiberCollection
```

Comment:
Can we put "Linear" or "Array" or something in the name to remind people of perf characteristics?